sdk/core/azure-core/inc/azure/core/internal/diagnostics/log.hpp

```diff
@@ -50,6 +50,19 @@ namespace Azure { namespace Core { namespace Diagnostics { namespace _internal {
       LoggerStringBuffer m_stringBuffer;
     };
 
+    class Stream {
+    public:
+      Stream(Logger::Level level) : m_stream(GetStream(level)) {}
+      ~Stream() { m_stream.flush(); }
+      Stream(Stream const&) = delete;
+      Stream& operator=(Stream const&) = delete;
+
+      template <typename T> std::ostream& operator<<(T val) { return m_stream << val; }
+
+    private:
+      LoggerStream& m_stream;
+    };
+
```

Comment:
What's the purpose of this class? Please add comments. It seems to be very closely tied to `LoggerStream` with a custom dtor for flushing and the `<<` operator. Could we just merge them both and keep one class, if the separation isn't critical? It doesn't seem to buy us much, unless I am missing some rationale for the separation.

Also, are we ok with a class name as generic as `Stream`.

Reply:
This class is how you use the Log class as an iostream target. You say `Log::Stream(<level>) << "Test";` 

There doesn't seem to be any place to add documentation on how to use the Log class, otherwise I would have added it there.